### DISCUSSION TITLE
I can't get any data through MQTT.

### DISCUSSION THREAD

**Discussion by socekin** [teslamate-org/teslamate#4443] [Q&A]
I installed TeslaMate on the server 192.168.2.3 using the default Docker configuration, but I couldn't get any topic values through MQTT.

I tested it locally and MQTT is connectable.
```
nc -zv 192.168.2.3 1883  
Connection to 192.168.2.3 port 1883 [tcp/ibm-mqisdp] succeeded!
```

However, I couldn't get any data when subscribing to the topic. Does anyone know what the problem is? Thanks.

`mosquitto_sub -h 192.168.2.3 -p 1883 -t "teslamate/cars/1/tpms_pressure_fl"` couldn't get any data.

In the teslamate MQTT container, I saw the following log:

![CleanShot 2024-12-16 at 23 36 19@2x](https://github.com/user-attachments/assets/ebb879ac-62c3-40b7-854a-c341b4342748)

**Comment by SupahTed** [comment]
Having the same problem. Did you ever get it solved?

**Reply by arcastro** [reply]
if you're able to connect, but not seeing any data, try subscribing to all topics to debug first:

you can docker exec into your MQTT container and run
* `mosquitto_sub -h localhost -t '#' -v`

The wildcard `#` tells it to subscribe to all topics.

Some, but not all, messages are published to MQTT with the retain flag set to true. The last published message for these topics are retained and always sent immediately to new subscribers. You'll see these messages upon connecting and subscribing above.

Others, without the retain flag, are only delivered real-time when the message is published, so you may not see this data unless you happen to already be connected and subscribed at the time that particular field changes to a new state.

**Reply by socekin** [reply]
> Having the same problem. Did you ever get it solved?

I obtained the MQTT data using Node-RED.
